src/Stream.php

```diff
@@ -676,6 +713,55 @@ public function toArray(): array
         return $result;
     }
 
+    /**
+     * Writes iterable source to the file line by line.
+     *
+     * Elements of the iterable source must be stringifiable.
+     *
+     * @param string $path
+     *
+     * @return void
+     *
+     * @throws Exceptions\FileException
+     */
+    public function toFileLines(string $path): void
+    {
+        $file = FileHelper::openToWrite($path);
+
+        /** @var string $line */
+        foreach ($this->iterable as $line) {
+            fputs($file, $line);
+        }
+    }
+
+    /**
+     * Writes iterable source to the file line by line.
+     *
+     * Elements of the iterable source must be array of scalars.
+     *
+     * @param string $path
+     * @param string $separator
+     * @param string $enclosure
+     * @param string $escape
+     *
+     * @return void
+     *
+     * @throws FileException
+     */
+    public function toCsvFile(
+        string $path,
```

Comment:
Question: The read file methods are taking a `resource` handle as input. For consistency, wouldn't it make sense to also take in a resource handle for the write methods as well? It also avoids dealing with the file system issues of opening a file handle and having it fail.

Reply:
I thought about such a solution, but I was confused that the user would have to open and close file resources on his own.

I think that from the point of view of the client, this is quite inconvenient.

Reply:
BTW we can add a new method `toResource()` that will take a resource as input.

Reply:
Opening and passing around a file handle or database handle or curl handle, etc. are common practices. It is also consistent with the other methods. I recommend taking in a file handle resource. If the user has not already opened the file handle for writing ahead of time, they could easily just inline the `fopen` within the `toFile()` stream method call.

Reply:
OK. [Done](https://github.com/markrogoyski/itertools-php/pull/19/commits/b0b7be2cee9ca39a029f25fe2faf742091a2d396).